frontend/src/components/Project/CreateProject.tsx

```diff
@@ -0,0 +1,561 @@
+    title: string;
+    description: string;
+    status: string;
+    serviceType: string;
+    leadSource: string;
+    leadCredit: string;
+    leadRanking: string;
+    followUpDate: string;
+    deadLineDate: string;
+    extraNotes: string;
+    email: string;
+    saleDescription: string;
+    saleValue: number | string;
+    assignee: Object[] | undefined;
+  }
+
+  interface EditFetchData {
+    id: number;
+    businessId: string;
+    createdDate: string;
+    deadlineDate: string;
+    description: string;
+    email: string;
+    extraNotes: string;
+    followUpDate: string;
+    leadCredit: string;
+    leadRanking: string;
+    leadSource: string;
+    sale: SaleCreationDTO;
+    serviceType: string;
+    status: string;
+    title: string;
+  }
+
+  const formik: FormikProps<CreateProjectFormData> = useFormik<CreateProjectFormData>({
+    onReset: () => {},
+    initialValues: {
+      title: '',
+      description: '',
+      status: '',
+      serviceType: '',
+      leadSource: '',
+      leadCredit: '',
+      leadRanking: '',
+      followUpDate: date.toISOString().split('T')[0],
+      deadLineDate: date.toISOString().split('T')[0],
+      extraNotes: '',
+      email: '',
+      saleDescription: '',
+      saleValue: '',
+      assignee: [],
+    },
+    onSubmit: async (values) => {
+      if (editState === 'true') {
+        submitEdit();
+      } else {
+        const response: AxiosResponse<any> = await createProject({
+          project: {
+            title: values.title,
+            description: values.description,
+            status: values.status,
+            serviceType: values.serviceType,
+            leadSource: values.leadSource,
+            leadCredit: values.leadCredit,
+            leadRanking: values.leadRanking,
+            deadlineDate: values.deadLineDate,
+            followUpDate: values.followUpDate,
+            extraNotes: values.extraNotes,
+            email: values.email,
+            businessId: 1,
+            assignee: values.assignee,
+          },
+          sale: {
+            amount: values.saleValue,
+            description: values.saleDescription,
+            dueDate: values.deadLineDate,
+          },
+        });
+        if (response.status === 201) {
+          setCreated(true);
+          history.push('/projects');
+        }
+      }
+    },
+    validationSchema: createProjectFromSchema,
+  });
+
+  const Input = styled('input')({
+    display: 'none',
+  });
+
+  const [employeeList, setEmployeeList] = useState<Array<{ label: string }>>([]);
+  const [assignee, setAssignee] = useState<any>([]);
+  const [clientList, setClientList] = useState<any>([]);
+  const [clientLoading, setClientLoading] = useState<boolean>(false);
+  const [assigneeLoading, setAssigeeLoading] = useState<boolean>(false);
+  const [editState, setEditState] = useState<string>(edit);
+  const [disabled, setDisabled] = useState<boolean>(false);
+
+  interface workson {
```

Comment:
move the interface outside the component.

Reply:
ok